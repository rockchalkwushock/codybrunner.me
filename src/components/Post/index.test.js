import React from 'react'
import { shallow } from 'enzyme'

import SitePost from './'

const ctx = {
  next: '/posts/2',
  prev: '/posts/1',
  slug: '/turdsspace/'
}
const meta = {
  disqusShortname: 'turdf',
  siteUrl: 'turdferguson.com'
}
const post = {
  excerpt: 'blah blah blah',
  fields: {
    slug: '/turdsspace/'
  },
  frontmatter: {
    date: 'February 03, 2018',
    draft: false,
    tags: ['JavaScript', 'fp', 'oop'],
    title: 'All About JS'
  },
  html: 'post body',
  timeToRead: 17,
  wordCount: {
    words: 567
  }
}

describe('Component: <SitePost />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<SitePost ctx={ctx} meta={meta} post={post} />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should have matching html', () => {
    expect(
      wrapper.find('Content').prop('dangerouslySetInnerHTML').__html
    ).toEqual('post body')
  })
})
