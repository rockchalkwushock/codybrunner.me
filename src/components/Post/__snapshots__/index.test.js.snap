// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <SitePost /> should render correctly 1`] = `
<Post
  draft={false}
>
  <SEO
    post={
      Object {
        "excerpt": "blah blah blah",
        "fields": Object {
          "slug": "/turdsspace/",
        },
        "frontmatter": Object {
          "date": "February 03, 2018",
          "draft": false,
          "tags": Array [
            "JavaScript",
            "fp",
            "oop",
          ],
          "title": "All About JS",
        },
        "html": "post body",
        "timeToRead": 17,
        "wordCount": Object {
          "words": 567,
        },
      }
    }
    postSeo={true}
    site={
      Object {
        "disqusShortname": "turdf",
        "siteUrl": "turdferguson.com",
      }
    }
  />
  <PostHeader
    date="February 03, 2018"
    time={17}
    title="All About JS"
    words={567}
  />
  <Content
    dangerouslySetInnerHTML={
      Object {
        "__html": "post body",
      }
    }
  />
  <Tags
    tags={
      Array [
        "JavaScript",
        "fp",
        "oop",
      ]
    }
  />
  <Pagination
    next={Object {}}
    prev={Object {}}
    slug="/turdsspace/"
  />
  <Share
    title="All About JS"
    url="turdferguson.com/turdsspace/"
  />
</Post>
`;
